phpBB/phpbb/kernel.php

```diff
@@ -0,0 +1,329 @@
+		return $this->php_ext;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function get_container()
+	{
+		return $this->container;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function get_start_time()
+	{
+		return $this->debug ? $this->start_time : -INF;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function get_cache_dir()
+	{
+		return $this->root_dir . '/cache/' . $this->environment;
+	}
+
+	/**
+	 * @param class_loader[] $autoloaders
+	 */
+	public function set_autoloaders(array $autoloaders)
+	{
+		$this->autoloaders = $autoloaders;
+	}
+
+	/**
+	 * Initializes the service container.
+	 *
+	 * The cached version of the service container is used when fresh, otherwise the
+	 * container is built.
+	 */
+	protected function initializeContainer()
+	{
+		try
+		{
+			$container_builder = new container_builder($this->root_dir, $this->php_ext);
+			$this->container = $container_builder
+				->with_cache_dir($this->get_cache_dir())
+				->with_environment($this->environment)
+				->with_config($this->config_file)
+				->get_container()
+			;
+
+			global $phpbb_container;
+			$phpbb_container = $this->container;
+		}
+		catch (InvalidArgumentException $e)
+		{
+			if (!$this->debug)
+			{
+				trigger_error(
+					'The requested environment ' . $this->environment . ' is not available.',
+					E_USER_ERROR
+				);
+			}
+			else
+			{
+				throw $e;
+			}
+		}
+	}
+
+	/**
+	 * Initialize stuff after loading the container
+	 */
+	protected function initializePostContainer()
+	{
+		foreach ($this->autoloaders as $autoloader)
+		{
+			$autoloader->set_cache($this->container->get('cache.driver'));
+		}
+
+		// In case $phpbb_adm_relative_path is not set (in case of an update), use the default.
+		global $phpbb_adm_relative_path, $phpbb_admin_path;
+		$phpbb_adm_relative_path = (isset($phpbb_adm_relative_path)) ? $phpbb_adm_relative_path : 'adm/';
+		$phpbb_admin_path = (defined('PHPBB_ADMIN_PATH')) ? PHPBB_ADMIN_PATH : $this->root_dir . $phpbb_adm_relative_path;
+
+		$this->loadFunctions();
+
+		require($this->root_dir . 'includes/compatibility_globals.' . $this->php_ext);
+
+		$this->load_hook_handler();
+	}
+
+	/**
+	 * Loads the functions files
+	 */
+	protected function loadFunctions()
+	{
+		require_once($this->root_dir . 'includes/functions.' . $this->php_ext);
+		require_once($this->root_dir . 'includes/functions_content.' . $this->php_ext);
+		require_once($this->root_dir . 'includes/functions_compatibility.' . $this->php_ext);
+		require_once($this->root_dir . 'includes/constants.' . $this->php_ext);
+		require_once($this->root_dir . 'includes/utf/utf_tools.' . $this->php_ext);
+	}
+
+	/**
+	 * Register phpBB's legacy hook handler
+	 *
+	 * @deprecated To be removed in 4.0
+	 */
+	protected function load_hook_handler()
+	{
+		global $phpbb_hook;
+
+		require($this->root_dir . 'includes/hooks/index.' . $this->php_ext);
+		$phpbb_hook = new \phpbb_hook(array('exit_handler', 'phpbb_user_session_handler', 'append_sid', ['template', 'display']));
```

Comment:
you're a crazy guy, going into this PR :p but thanks :)

About the old array syntax... I think I just copy/paste this part of the code :) (the coding style allows both syntax for now)